tests/kernel/smp/src/main.c

```diff
@@ -83,9 +84,48 @@ void test_main(void)
 		}
 	}
 
-	if (ok) {
-		TC_END_REPORT(TC_PASS);
-	} else {
-		TC_END_REPORT(TC_FAIL);
-	}
+	zassert_true(ok, "SMP test failed");
+}
+
+static void child_fn(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p3);
+	int parent_cpu_id = (int)p1;
+	struct k_thread *parent_thread_id = (struct k_thread *)p2;
+
+	zassert_true(parent_cpu_id != _arch_curr_cpu()->id &&
+		     _kernel.cpus[parent_cpu_id].current
+		     == parent_thread_id,
+		     "Parent isn't on other core");
 }
+
+/**
+ * @brief Verify CPU IDs of threads in SMP
+ *
+ * @details Verify whether thread running on other core is
+ * parent thread from child thread
+ */
+void test_cpu_id_threads(void)
+{
+	int parent_cpu_id = _arch_curr_cpu()->id;
+
+	k_thread_create(&t2, t2_stack, T2_STACK_SIZE, child_fn,
+			(void *)parent_cpu_id, &t2,
+			NULL, K_PRIO_PREEMPT(2), 0, K_NO_WAIT);
```

Comment:
This looks racy.  What happens if test_cpu_id_threads finishes and exits (which it does synchronously as soon as k_thread_create() returns) before child_fn starts running on the other CPU?  Seems not at all unlikely, and will produce a spurious failure.

You need synchronization here.  Block on a semaphore that is given in child_fn, for example.